### PR TITLE
[WFLY-11352] WildFly registers multiple distinct drivers for current MySQL driver jar - rework

### PR DESCRIPTION
**Pull request by gaol**
Jira: https://issues.jboss.org/browse/WFLY-11352

This is rework for WFLY-11352, because it caused a regression in case of multiple drivers defined in the JDBC driver.

The proposed fix is trying to start an extra DriverService with DeploymentUnit name as the driver-name for the first one as well if there are more than one Driver classes defined in META-INF/services/java.sql.Driver, so that both following command can work:

> /subsystem=datasources:get-installed-driver(driver-name=mysql-connector-java-5.1.46.jar)
> /subsystem=datasources:get-installed-driver(driver-name=mysql-connector-java-5.1.46.jar_com.mysql.jdbc.Driver_5_1)

Please make sure your PR meets the following requirements:
- [ ] Pull Request title is properly formatted: `[WFLY-XYZ] Subject` or `WFLY-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue(s)
- [ ] Pull Request contains description of the issue(s)
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Attached commits represent units of work and are properly formatted

### CODE REVIEW

**Comment by jamezp** [comment]
@maeste What do you think about this?

**Comment by jamezp** [comment]
This seems to work with the expected results for me. @simkam could you please have a look at this?

**Review by simkam — APPROVED** [review]
I ran some tests (including those that revealed regression in previous fix) and they didn't find any issue. LGTM

**Comment by jamezp** [comment]
We are going to end up going with #12153. However after that is merged, this PR can be rebased and we can wait for @maeste to approve it. So this could end up in WildFly 17 still.

**Comment by gaol** [comment]
rebased

**Review by maeste — APPROVED** [review]

**Pull request merged**
